### PR TITLE
[Restructure/plasma] estimator cleanup

### PR DESCRIPTION
**Pull request by wkerzendorf**
### :pencil: Description

**Type:** :beetle: `bugfix` | :rocket: `feature` | :biohazard: `breaking change` | :vertical_traffic_light: `testing` | :memo: `documentation` | :roller_coaster: `infrastructure`

Write a complete description of your changes, including the necessary context or any piece of information required to understand your work.

Also, link issues affected by this pull request by using the keywords: `close`, `closes`, `closed`, `fix`, `fixes`, `fixed`, `resolve`, `resolves` or `resolved`. 

### :pushpin: Resources

Examples, notebooks, and links to useful references.

### :vertical_traffic_light: Testing

How did you test these changes?

- [ ] Testing pipeline
- [ ] Other method (describe)
- [ ] My changes can't be tested (explain why)

### :ballot_box_with_check: Checklist

- [ ] I requested two reviewers for this pull request
- [ ] I updated the documentation according to my changes
- [ ] I built the documentation by applying the `build_docs` label

> **Note:** If you are not allowed to perform any of these actions, ping (@) a contributor.

### CODE REVIEW

File: benchmarks/opacities_opacity.py [hunk 1]
```diff
@@ -28,7 +29,7 @@ class BenchmarkMontecarloMontecarloNumbaOpacities(BenchmarkBase):
         }
     )
     def time_compton_opacity_calculation(self, electron_number_density, energy):
-        calculate_opacity.compton_opacity_calculation(
+        tardis.opacities.compton_opacity_calculation.compton_opacity_calculation(
```
Comment (andrewfullard) [inline benchmarks/opacities_opacity.py:32]: `import as` perhaps?

Comment (andrewfullard) [reply benchmarks/opacities_opacity.py:32]: Or `from x import compton_opacity_calculation`

File: tardis/opacities/opacities.py [hunk 2]
```diff
@@ -278,7 +278,8 @@ def compton_opacity_partial(energy, fraction):
 
 @njit(**njit_dict_no_parallel)
 def compton_opacity_calculation(energy, electron_density):
```
Comment (andrewfullard) [inline tardis/opacities/opacities.py:280]: This should have been moved?

File: tardis/plasma/base.py [hunk 3]
```diff
@@ -284,11 +292,10 @@ def write_to_dot(self, fname, args=None, latex_label=True):
             enables/disables writing LaTeX equations and
             edge labels into the file.
         """
-
         try:
-            import pygraphviz
+            pass
```
Comment (andrewfullard) [inline tardis/plasma/base.py:296]: ??

File: tardis/plasma/radiation_field/planck_rad_field.py [hunk 4]
```diff
@@ -72,3 +72,60 @@ def calculate_mean_intensity(self, nu: Union[u.Quantity, np.ndarray]):
...
             nu[np.newaxis].T, self.temperature
         )
+
+    def to_planckian_radiation_field(self):
+        return PlanckianRadiationField(self.temperature)
+
+
+class PlanckianRadiationField:
+    """
+    Represents the state of a dilute thermal radiation field.
```
Comment (andrewfullard) [inline tardis/plasma/radiation_field/planck_rad_field.py:82]: Is this dilute?

File: tardis/transport/montecarlo/estimators/dilute_blackbody_properties.py [hunk 5]
```diff
@@ -18,11 +21,20 @@
 ).cgs.value
 
 
-class MCDiluteBlackBodyRadFieldSolver:
-    def __init__(self) -> None:
-        pass
+class MCRadiationFieldPropertiesSolver:
```
Comment (andrewfullard) [inline tardis/transport/montecarlo/estimators/dilute_blackbody_properties.py:24]: Is this meant to be a generic solver class? The file should be renamed in that case.

**Review by andrewfullard — CHANGES_REQUESTED** [review]

File: tardis/transport/montecarlo/estimators/dilute_blackbody_properties.py (hunk 5 above)
Comment (wkerzendorf) [reply tardis/transport/montecarlo/estimators/dilute_blackbody_properties.py:24]: I don't know - it had that name before.

File: tardis/opacities/opacities.py (hunk 2 above)
Comment (wkerzendorf) [reply tardis/opacities/opacities.py:280]: nope - I decided against it for this PR. But in general yes.

File: tardis/plasma/radiation_field/planck_rad_field.py (hunk 4 above)
Comment (wkerzendorf) [reply tardis/plasma/radiation_field/planck_rad_field.py:82]: nope, why?

File: benchmarks/opacities_opacity.py (hunk 1 above)
Comment (wkerzendorf) [reply benchmarks/opacities_opacity.py:32]: this should not have moved

File: tardis/plasma/radiation_field/planck_rad_field.py (hunk 4 above)
Comment (andrewfullard) [reply tardis/plasma/radiation_field/planck_rad_field.py:82]: Then the docstring is wrong ;)

File: tardis/transport/montecarlo/tests/test_opacities.py [hunk 6]
```diff
@@ -1,8 +1,8 @@
 import numpy.testing as npt
 import pytest
+from tardis.opacities.opacities import compton_opacity_calculation
 
 from tardis.opacities.opacities import (
-    compton_opacity_calculation,
```
Comment (andrewfullard) [inline tardis/transport/montecarlo/tests/test_opacities.py:5]: Pointless change?

**Review by andrewfullard — CHANGES_REQUESTED** [review]

**Comment by andrewfullard** [comment]
https://github.com/tardis-sn/tardis/actions/runs/10115538957 docs passed at least.

**Comment by DeerWhale** [comment]
> Benchmarks still aren't quite working on PRs here so we can't check that.

The benchmark is now working after merging in #2747.

**Review by andrewfullard — APPROVED** [review]

File: tardis/energy_input/gamma_packet_loop.py [hunk 7]
```diff
@@ -19,9 +19,11 @@
     doppler_factor_3d,
     get_index,
 )
+from tardis.opacities.compton_opacity_calculation import (
```
Comment (DeerWhale) [inline tardis/energy_input/gamma_packet_loop.py:22]: Should here be "tardis.opacities.opacities" instead?

File: tardis/plasma/properties/continuum_processes/fast_array_util.py [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+    Returns
+    -------
+    numpy.ndarray, dtype float
+        The result of cumulative integration of f along x
+    """
+    integ = (np.diff(x) * (f[1:] + f[:-1]) / 2.0).cumsum()
+    return integ / integ[-1]
+
+
+@njit(**njit_dict)
```
Comment (Knights-Templars) [inline tardis/plasma/properties/continuum_processes/fast_array_util.py:31]: Please make an issue

File: tardis/plasma/properties/continuum_processes/photo_ion_rate_coeff.py [hunk 9]
```diff
@@ -0,0 +1,15 @@
+from tardis.plasma.properties.base import Input
+from tardis.plasma.properties.continuum_processes.rates import H
```
Comment (Knights-Templars) [inline tardis/plasma/properties/continuum_processes/photo_ion_rate_coeff.py:2]: This is not needed

File: tardis/energy_input/gamma_ray_estimators.py [hunk 10]
```diff
@@ -7,9 +7,11 @@
     angle_aberration_gamma,
     doppler_factor_3d,
 )
+from tardis.opacities.compton_opacity_calculation import (
+    compton_opacity_calculation,
+)
```
Comment (andrewfullard) [inline tardis/energy_input/gamma_ray_estimators.py:12]: Unnecessary change

File: tardis/opacities/macro_atom/continuum_processes/collisional_ion_trans_prob.py [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+
+from tardis import constants as const
+from tardis.plasma.properties.base import (
+    TransitionProbabilitiesProperty,
+)
+from tardis.plasma.properties.continuum_processes.rates import (
+    IndexSetterMixin,
+)
+
+H = const.h.cgs.value
```
Comment (DeerWhale) [inline tardis/opacities/macro_atom/continuum_processes/collisional_ion_trans_prob.py:11]: If there will be a place to hold all the constants soon, this should be updated as well.

File: tardis/transport/montecarlo/estimators/dilute_blackbody_properties.py (hunk 5 above)
Comment (andrewfullard) [reply tardis/transport/montecarlo/estimators/dilute_blackbody_properties.py:24]: Please rename the file to match what the class is doing

File: tardis/transport/montecarlo/montecarlo_transport_state.py [hunk 12]
```diff
@@ -63,6 +63,7 @@ def __init__(
         rpacket_tracker=None,
         vpacket_tracker=None,
     ):
+        self.time_explosion = time_explosion
```
Comment (Knights-Templars) [inline tardis/transport/montecarlo/montecarlo_transport_state.py:66]: Duplicate!

File: tardis/energy_input/gamma_packet_loop.py (hunk 7 above)
Comment (wkerzendorf) [reply tardis/energy_input/gamma_packet_loop.py:22]: done

**Review by andrewfullard — APPROVED** [review]

**Review by jvshields — APPROVED** [review]
